tools/run-tests.py

```diff
@@ -29,6 +30,11 @@
 Options = collections.namedtuple('Options', ['name', 'build_args', 'test_args'])
 Options.__new__.__defaults__ = ([], [])
 
+SkippedOptions = collections.namedtuple('SkippedOptions', ['name', 'build_args', 'test_args', 'skip'])
```

Comment:
What if we do something like this?
```python
Options = collections.namedtuple('Options', ['name', 'build_args', 'test_args', 'skip'])
Options.__new__.__defaults__ = ([], [], False)

def skip_if(condition, desc):
    return desc if condition else False
```

Then we don't need the extra `SkippedOptions` and the new `Options` could be used like this:
```python
Options('buildoption_test-cpointer_32bit',
        ['--compile-flag=-m32', '--cpointer-32bit=on', '--system-allocator=on']),
        skip=skip_if(
            platform.system() != 'Linux' or (platform.machine() != 'i386' and platform.machine() != 'x86_64'),
            '-m32 is only supported on x86[-64]-linux')
        )
)
```
Of course then we need to checking the `skip` with `hasattr` is not enough and the skip description should be a valid non-empty string, but that can be checked (at least the length) in the `skip_if` method to ensure the developers put there meaningful text about by this option is skipped.

What do you think?